controllers/btpoperator_controller.go

```diff
@@ -1212,3 +1326,518 @@ func (r *BtpOperatorReconciler) IsForceDelete(cr *v1alpha1.BtpOperator) bool {
 	}
 	return cr.Labels[forceDeleteLabelKey] == "true"
 }
+
+// *[]*unstructured.Unstructured is required for changing the length of slice in certificates regeneration
+func (r *BtpOperatorReconciler) reconcileCertificates(ctx context.Context, resourcesToApply *[]*unstructured.Unstructured) error {
+	logger := log.FromContext(ctx)
+	logger.Info("certificates reconciliation started")
+
+	certificatesRegenerationDone, err := r.ensureCertificatesExists(ctx, resourcesToApply)
+	if err != nil {
+		return err
+	}
+	if certificatesRegenerationDone {
+		return nil
+	}
+
+	certificatesRegenerationDone, err = r.ensureSecretsDataIsSet(ctx, resourcesToApply)
+	if err != nil {
+		return err
+	}
+	if certificatesRegenerationDone {
+		return nil
+	}
+
+	certificatesRegenerationDone, err = r.ensureCertificatesAreCorrectlyStructured(ctx, resourcesToApply)
+	if err != nil {
+		return err
+	}
+	if certificatesRegenerationDone {
+		return nil
+	}
+
+	certificatesRegenerationDone, err = r.ensureCertificatesHaveValidExpiration(ctx, resourcesToApply)
+	if err != nil {
+		return err
+	}
+	if certificatesRegenerationDone {
+		return nil
+	}
+
+	certificatesRegenerationDone, err = r.ensureCertificatesAreCorrectSigned(ctx, resourcesToApply)
+	if err != nil {
+		return err
+	}
+	if certificatesRegenerationDone {
+		return nil
+	}
+
+	if err := r.reconcileWebhooksConfigurations(ctx, resourcesToApply, nil); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *BtpOperatorReconciler) ensureCertificatesExists(ctx context.Context, resourcesToApply *[]*unstructured.Unstructured) (bool, error) {
+	logger := log.FromContext(ctx)
+	caSecretExists, err := r.checkIfSecretExists(ctx, CaSecret)
+	if err != nil {
+		return false, err
+	}
+	if !caSecretExists {
+		logger.Info("CA secret with cert doesn't exists")
+		if err := r.doFullCertificatesRegeneration(ctx, resourcesToApply); err != nil {
+			return false, err
+		}
+		return true, nil
+	}
+	logger.Info("CA secret exists")
+	webhookSecretExists, err := r.checkIfSecretExists(ctx, WebhookSecret)
+
+	if err != nil {
+		return false, err
+	}
+	if !webhookSecretExists {
+		logger.Info("webhook secret with cert does not exists")
+		if err := r.doPartialCertificatesRegeneration(ctx, resourcesToApply); err != nil {
+			return false, err
+		}
+		return true, nil
+	}
+	logger.Info("webhook secret exists")
+	return false, nil
+}
+
+func (r *BtpOperatorReconciler) ensureSecretsDataIsSet(ctx context.Context, resourcesToApply *[]*unstructured.Unstructured) (bool, error) {
+	caSecretData, err := r.getDataFromSecret(ctx, CaSecret)
+	caSecretDataIncorrect := false
+	_, err = r.getValueByKey(r.buildKeyNameWithExtension(CaSecretDataPrefix, CertificatePostfix), caSecretData)
+	if err != nil {
```

Comment:
`caSecretDataIncorrect := err != nil` would do. (3 lines less)

Reply:
At the beginning `caSecretDataIncorrect` is `false`.
The logic states that if err != nil `caSecretDataIncorrect` is set to `true`. Otherwise it is still `false`.
So this is equivalent of `caSecretDataIncorrect := err != nil`.
Nothing is overwritten.

In subsequent `ifs` you can use:
`caSecretDataIncorrect = caSecretDataIncorrect || err != nil`